app/helpers/katello/hosts_and_hostgroups_helper.rb

```diff
@@ -68,8 +71,12 @@ def content_views_for_host(host, options)
         include_blank = '<option></option>'
       end
 
-      content_view = content_view(host)
+      selected_host_group = options.fetch(:selected_host_group, nil)
       lifecycle_environment = lifecycle_environment(host)
+      lifecycle_environment = lifecycle_environment(selected_host_group) if lifecycle_environment.nil? && selected_host_group.present?
```

Comment:
I'm sensing some repeated logic between this and https://github.com/Katello/katello/pull/5814/files#diff-a29a1f54f7213ac056cf5af0631d90fcR45